Fixes for newer haxe versions and compile scripts

crowplexus opened the pull request:
thanks @Rudyrue for the batch script cuz I'm stupid yeye

* Haxe 4.3 and higher versions now compile properly
* We included new scripts to compile the game easily

crowplexus committed:
fix haxe 4.3+ and add new compile scripts

Co-Authored-By: Rudyrue <73214127+Rudyrue@users.noreply.github.com>

Rudyrue commented:
yipe............................................................................................................

crowplexus committed:
Update README.md

Haxe 4.2.5 *specifically* is no longer needed, you can use the latest versions now

Stilic commented:
Why isn't the setup script parsing the hmm.json file instead of a separate deps file?
This would allow the usage of the old method for people that have this method working for them.

crowplexus committed:
keep hmm method of install

crowplexus committed:
Merge branch 'hmm-but-awesome' of https://github.com/CrowPlexus/FNF-PsychEngine into hmm-but-awesome

crowplexus committed:
modify setup files so they directly install libraries

crowplexus committed:
remove base game markdown files

they aren't really accurate to psych anyways
